# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/loader/ModuleLoadRequest.h
# Commit: f3f57e90febf
# Full Hash: f3f57e90febfd10f320eb3b28dee3e938ca62fd1
# Author: Yoshi Cheng-Hao Huang <allstars.chh@gmail.com>
# Date: 2025-10-03 20:53 +0000
# Regressor Bug: 1968890
# File Overlap Count: 8
# Overlapping Files: dom/script/ScriptLoadContext.cpp, dom/script/ModuleLoader.cpp, js/loader/ModuleLoadRequest.cpp, js/loader/ModuleLoaderBase.cpp, dom/workers/loader/WorkerModuleLoader.cpp
# Description:
#   Bug 1968890 - Part 1: Dynamic import is not a top-level request. r=jonco,dom-worker-reviewers,edenchuang
# ==============================================================================

diff -r d07fc2f2bc30 -r f3f57e90febf js/loader/ModuleLoadRequest.h
--- a/js/loader/ModuleLoadRequest.h	Fri Oct 03 20:45:33 2025 +0000
+++ b/js/loader/ModuleLoadRequest.h	Fri Oct 03 20:53:28 2025 +0000
@@ -40,15 +40,13 @@
   using SRIMetadata = mozilla::dom::SRIMetadata;
 
   enum class Kind {
-    // Top-level modules, not imported statically or dynamically..
+    // Top-level modules, not imported statically nor dynamically.
     TopLevel,
 
     // Modules imported statically with `import` declarations.
     StaticImport,
 
     // Modules imported dynamically with dynamic `import()`.
-    // This is actually also a top-level module, but this should be used for
-    // dynamic imports.
     DynamicImport,
   };
 
@@ -59,9 +57,9 @@
                     LoadContextBase* aContext, Kind aKind,
                     ModuleLoaderBase* aLoader, ModuleLoadRequest* aRootModule);
 
-  bool IsTopLevel() const override { return mIsTopLevel; }
-
-  bool IsDynamicImport() const { return mIsDynamicImport; }
+  bool IsTopLevel() const override { return mKind == Kind::TopLevel; }
+  bool IsStaticImport() const { return mKind == Kind::StaticImport; }
+  bool IsDynamicImport() const { return mKind == Kind::DynamicImport; }
 
   bool IsErrored() const;
 
@@ -113,15 +111,11 @@
     mReferrerPolicy = aReferrerPolicy;
   }
 
-  // Is this a request for a top level module script or an import?
-  const bool mIsTopLevel;
+  const Kind mKind;
 
   // Type of module (JavaScript, JSON)
   const ModuleType mModuleType;
 
-  // Is this the top level request for a dynamic module import?
-  const bool mIsDynamicImport;
-
   // Pointer to the script loader, used to trigger actions when the module load
   // finishes.
   RefPtr<ModuleLoaderBase> mLoader;